install/helm/agones/templates/service/allocation.yaml

```diff
@@ -26,31 +26,46 @@ metadata:
     chart: {{ template "agones.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
-{{- if .Values.agones.allocator.http.annotations }}
+{{- if .Values.agones.allocator.service.annotations }}
   annotations:
-{{ toYaml .Values.agones.allocator.http.annotations | indent 4 }}
+{{ toYaml .Values.agones.allocator.service.annotations | indent 4 }}
 {{- end }}
 spec:
   selector:
     multicluster.agones.dev/role: allocator
   ports:
-    - port: {{ .Values.agones.allocator.http.port }}
+{{- if .Values.agones.allocator.service.http.enabled }}
```

Comment:
The logic here is similar to what is in `cmd/allocator/main.go` but is a bit harder to read due to the templating language.

There are 4 cases to cover:
1. If http & grpc are both enabled and use the same port, only configure the http port
1. If http & grpc are both enabled and use different ports, configure the http port and the grpc port
1. If only http is enabled, only configure the http port
1. If only grpc is enabled, only configure the grpc port

The logic to cover the 4 cases has three branches:

1. If the http port is enabled, then use it (covers cases 1-3 & skipping covers 4)
   1. If the grpc port is also enabled and it is different from the http port, use it as well (case 2)
1. If the http port is disabled but the grpc port is enabled, use it (case 4)

Locally I tested this by modifying my `values.yaml` file and running `make gen-install` to see the resulting yaml and verify that the correct number or ports were defined (1 or 2) with the correct values.